dist/src/main/java/io/camunda/application/commons/console/ping/PingConsoleTask.java

```diff
@@ -0,0 +1,146 @@
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.application.commons.console.ping;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.application.commons.console.ping.PingConsoleConfiguration.ConsolePingConfiguration;
+import io.camunda.service.ManagementServices;
+import io.camunda.zeebe.util.VisibleForTesting;
+import io.camunda.zeebe.util.retry.RetryConfiguration;
+import io.camunda.zeebe.util.retry.RetryDecorator;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.time.Duration;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PingConsoleTask implements Runnable {
+
+  public static final DateTimeFormatter DATE_TIME_FORMATTER =
+      DateTimeFormatter.ofPattern("uuuu-MM-dd'T'HH:mm:ss'Z'").withZone(ZoneOffset.UTC);
+  private static final Logger LOGGER = LoggerFactory.getLogger(PingConsoleTask.class);
+  private static final int NUMBER_OF_MAX_RETRIES = 10;
+  private static final int RETRY_DELAY_MULTIPLIER = 2;
+  private static final Duration MAX_RETRY_DELAY = Duration.ofMinutes(2);
+  public final HttpClient client;
+  private final ManagementServices managementServices;
+  private final ConsolePingConfiguration pingConfiguration;
+  private final RetryDecorator retryDecorator;
+
+  @VisibleForTesting
+  public PingConsoleTask(
+      final ManagementServices managementServices,
+      final ConsolePingConfiguration pingConfiguration,
+      final HttpClient client) {
+    this.managementServices = managementServices;
+    this.pingConfiguration = pingConfiguration;
+    this.client = client;
+    final var retryConfiguration = new RetryConfiguration();
+    retryConfiguration.setMaxRetries(NUMBER_OF_MAX_RETRIES);
+    retryConfiguration.setRetryDelayMultiplier(RETRY_DELAY_MULTIPLIER);
+    retryConfiguration.setMaxRetryDelay(MAX_RETRY_DELAY);
+    retryDecorator = new RetryDecorator();
+  }
+
+  public PingConsoleTask(
+      final ManagementServices managementServices,
+      final ConsolePingConfiguration pingConfiguration) {
+    this(managementServices, pingConfiguration, HttpClient.newHttpClient());
+  }
+
+  @Override
+  public void run() {
+    try {
+      final HttpRequest request =
+          HttpRequest.newBuilder()
+              .uri(URI.create(pingConfiguration.endpoint()))
+              .POST(HttpRequest.BodyPublishers.ofString(getLicensePayload()))
```

Comment:
🔧 Unless I'm missing something, the payload is actually constant (until the app is restarted with different configuration). Let's not fetch it every time.